prime-router/src/main/kotlin/Report.kt

```diff
@@ -195,7 +195,7 @@ class Report : Logging {
     /**
      * The number of items in the report
      */
-    val itemCount: Int
+    var itemCount: Int? = null
```

Comment:
This new Receive step does not parse the message. So the item count isn't known until the convert step.

Reply:
@arnejduranovic In my head null feels more accurate than zero since we don't know how many items there are. To me, putting `0` items says that the message has exactly zero items. Then when the convert step says the message has 3 items the lineage could be confusing.

Reply:
I wonder if it would make sense to split off a `submission` into its own class. Just a random thought so don't pay this too much mind.